sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -250,8 +251,14 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
    * table/view. This throws an `AnalysisException` when no `Table` can be found.
    */
   override def getTable(tableName: String): Table = {
-    val tableIdent = sparkSession.sessionState.sqlParser.parseTableIdentifier(tableName)
-    getTable(tableIdent.database.orNull, tableIdent.table)
+    try {
+      val ident = sparkSession.sessionState.sqlParser.parseTableIdentifier(tableName)
+      getTable(ident.database.orNull, ident.table)
+    } catch {
+      case e: org.apache.spark.sql.catalyst.parser.ParseException =>
+        val ident = sparkSession.sessionState.sqlParser.parseMultipartIdentifier(tableName)
```

Comment:
will `parseMultipartIdentifier` throw `ParseException`?

Reply:
No. For arbitrary sequence it can breaks that by dots to a Seq of string.

`sparkSession.sessionState.sqlParser.parseTableIdentifie` will throw exception when the input is not `b` or `a.b` thus beyond a table identifier which only considers DB and table name in OSS.

Reply:
That arbitrary sequence of identifier will be passed to SQL analyzer to resolve the table. If user for example give `a.b.c.d.e.f`, the analyzer will just say this is not found.